calculator_test.go

```diff
@@ -3,22 +3,196 @@ package calculator_test
 import (
 	"calculator"
 	"testing"
+	"math/rand"
+	"time"
+	
 )
 
 func TestAdd(t *testing.T) {
 	t.Parallel()
-	var want float64 = 4
-	got := calculator.Add(2, 2)
-	if want != got {
-		t.Errorf("want %f, got %f", want, got)
+	type testCase struct {
+		a, b float64
+		want float64
+		message string
```

Comment:
Good! It's a nice idea to put the `message` field first (let's call it `name` for brevity, and it's also the name of the test case, if you like). That way, when you list your test cases, the name comes first.